tianshou/data/batch.py

```diff
@@ -333,13 +314,17 @@ def __getitem__(self, index: Union[
         """Return self[index]."""
         if isinstance(index, str):
             return self.__dict__[index]
-        b = Batch()
-        for k, v in self.items():
-            if isinstance(v, Batch) and len(v.__dict__) == 0:
-                b.__dict__[k] = Batch()
-            else:
-                b.__dict__[k] = v[index]
-        return b
+        batch_items = self.items()
+        if len(batch_items) > 0:
+            b = Batch()
+            for k, v in batch_items:
+                if isinstance(v, Batch) and len(v.__dict__) == 0:
+                    b.__dict__[k] = Batch()
+                else:
+                    b.__dict__[k] = v[index]
+            return b
+        else:
+            raise IndexError("Cannot access item from empty Batch object.")
```

Comment:
They may lead to a strange behavior:
```
d=Batch(a=[1,2,3], b=dict(c=[3,4,5]), c={}) 
d[2] # no problem
d.c[2] # exception
```
I'm not sure this is ok.

Reply:
In my opinion it is the desired behavior since it means different things. Doing:
```
d=Batch(a=[1,2,3], b=dict(c=[3,4,5]), c={}) 
```
you are just "reserving" the key `c`. 

But as soon as you do `d.c`, you get an empty `Batch` and that's all. And as such, accessing an item from an empty Batch raises an exception. In those two cases, `Batch()` as a different meaning.